task-sdk/src/airflow/sdk/definitions/dag.py

```diff
@@ -557,9 +557,17 @@ def _validate_catchup(self, _, catchup: bool):
             raise ValueError("start_date is required when catchup=True")
 
     @tags.validator
-    def _validate_tags(self, _, tags: Collection[str]):
-        if tags and any(len(tag) > TAG_MAX_LEN for tag in tags):
-            raise ValueError(f"tag cannot be longer than {TAG_MAX_LEN} characters")
+    def _validate_tags(self, _, tags: Collection[str]) -> None:
+        if tags:
+            for tag in tags:
+                if len(tag) > TAG_MAX_LEN:
+                    from airflow.exceptions import AirflowException
+                    # Trim very long tag previews to keep error messages manageable
+                    tag_preview = tag[:30] + "..." if len(tag) > 30 else tag
+                    raise AirflowException(
```

Comment:
We don't want to introduce or use new cases of AirflowException `ValueError` is more appopriate:

> Raised when an operation or function receives an argument that has the right type but an inappropriate value

```suggestion
                    raise ValueError(
```